source/1_engine/rex_engine/src/cmdline.cpp

```diff
@@ -0,0 +1,205 @@
+        }
+      }
+
+      void add_argument(rsl::string_view arg)
+      {
+        const count_t equal_pos = arg.find('=');
+        rsl::string_view key    = "";
+        rsl::string_view value  = "";
+
+        // if the argument is of type -MyArg=Something
+        if(equal_pos != -1)
+        {
+          key   = arg.substr(0, equal_pos);
+          value = arg.substr(equal_pos + 1);
+        }
+        // if the argument is of type -EnableSomething
+        else
+        {
+          key   = arg;
+          value = "1"; // this is so we can easily convert to bool/int
+        }
+
+        auto cmd_it = rsl::find_if(g_command_line_args.cbegin(), g_command_line_args.cend(), [key](const Argument& cmdArg) { return key == cmdArg.name; });
+
+        if(cmd_it == g_command_line_args.cend())
+        {
+          REX_WARN(LogEngine, "Command {} passed in but it's not recognised as a valid command so will be ignored", arg);
+          return;
+        }
+
+        auto active_it = rsl::find_if(m_arguments.cbegin(), m_arguments.cend(), [key](const ActiveArgument& activeArg) { return activeArg.argument == key; });
+
+        if(active_it != m_arguments.cend())
+        {
+          REX_WARN(LogEngine, "Command {} was already passed in. passing the same argument multiple times is not supported. will be skipped", key);
+          return;
+        }
+
+        m_arguments.push_back({key, value});
+      }
+
+      bool verify_args(const Argument* args, count_t argCount) // NOLINT(readability-convert-member-functions-to-static)
+      {
+        for(count_t i = 0; i < argCount; ++i)
+        {
+          const Argument& lhs_arg = args[i];
+          for(count_t j = 0; j < argCount; ++j)
+          {
+            if(i == j)
+              continue;
+
+            const Argument& rhs_arg = args[j];
+            if(rsl::strincmp(lhs_arg.name.data(), rhs_arg.name.data(), lhs_arg.name.length()) == 0)
+            {
+              REX_ERROR(LogEngine, "This executable already has an argument for {} specified in 'g_command_line_args', please resolve the ambiguity", lhs_arg.name);
+              return false;
+            }
+          }
+        }
+
+        return true;
+      }
+
+    private:
+      rsl::vector<ActiveArgument> m_arguments;
+    };
+
+    rsl::unique_ptr<CommandLineArguments> g_cmd_line_args; // NOLINT(fuchsia-statically-constructed-objects, cppcoreguidelines-avoid-non-const-global-variables)
+
+    void init(rsl::string_view cmdLine)
+    {
+      g_cmd_line_args = rsl::make_unique<CommandLineArguments>(cmdLine);
+    }
+
+    void print_args()
+    {
+      REX_LOG(LogEngine, "Listing Command line arguments");
+
+      rsl::unordered_map<rsl::string_view, rsl::vector<Argument>> project_to_arguments;
+
+      for(Argument cmd: g_command_line_args)
+      {
+        project_to_arguments[cmd.module].push_back(cmd);
+      }
+
+      for(auto [project, cmds]: project_to_arguments)
```

Comment:
Not really a fan of structured bindings ... dunno how you feel about it but I rather have a struct.
But you can keep it.